src/http-api/index.js

```diff
@@ -22,56 +16,61 @@ function uriToMultiaddr (uri) {
   return `/ip4/${ipPort[0]}/tcp/${ipPort[1]}`
 }
 
-function HttpApi (repo) {
+function HttpApi (repo, config) {
   this.node = undefined
   this.server = undefined
 
-  this.start = (callback) => {
-    if (typeof repo === 'string') {
-      repo = new IPFSRepo(repo, {stores: Store})
-    }
+  this.log = debug('jsipfs:http-api')
+  this.log.error = debug('jsipfs:http-api:error')
 
-    let apiFilePath
+  if (typeof repo === 'string') {
+    repo = new IPFSRepo(repo)
+  }
+
+  this.start = (init, callback) => {
+    if (typeof init === 'function') {
+      callback = init
+      init = false
+    }
+    this.log('starting')
 
     series([
       (cb) => {
         // start the daemon
-        this.node = new IPFS({
-          repo: repo,
-          init: false,
-          start: true,
-          EXPERIMENTAL: {
-            pubsub: true
-          }
-        })
+        try {
+          this.node = new IPFS({
+            repo: repo,
+            init: init,
+            start: true,
+            config: config,
+            EXPERIMENTAL: {
+              pubsub: true
+            }
+          })
+        } catch (err) {
+          return cb(err)
+        }
```

Comment:
this can throw if someone screwed up in the implementation, resulting in silent fail of the tests